src/webui/www/private/searchplugins.html

```diff
@@ -0,0 +1,217 @@
+            paddingVertical: 0,
+            paddingHorizontal: 0,
+            width: 500,
+            height: 120
+        });
+    };
+
+    uninstallPlugin = function() {
+        var plugins = searchPluginsTable.selectedRowsIds().join('|');
+        var url = new URI('api/v2/search/uninstallPlugin');
+        var request = new Request({
+            url: url,
+            noCache: true,
+            method: 'post',
+            data: {
+                names: plugins,
+            }
+        }).send();
+    };
+
+    enablePlugin = function() {
+        var plugins = searchPluginsTable.selectedRowsIds();
+        var enable = true;
+        if (plugins && plugins.length)
+            enable = !getPlugin(plugins[0]).enabled;
+
+        var url = new URI('api/v2/search/enablePlugin');
+        var request = new Request({
+            url: url,
+            noCache: true,
+            method: 'post',
+            data: {
+                names: plugins.join('|'),
+                enable: enable
+            }
+        }).send();
+    };
+
+    checkForUpdates = function() {
+        var url = new URI('api/v2/search/updatePlugins');
+        var request = new Request({
+            url: url,
+            noCache: true,
+            method: 'post'
+        }).send();
+    };
+
+    calculateContextMenuOffsets = function() {
+        prevOffsetLeft = document.getElementById("searchPlugins").getBoundingClientRect().left;
+        prevOffsetTop = document.getElementById("searchPlugins").getBoundingClientRect().top;
+
+        return {
+            x: -(prevOffsetLeft + 20),
+            y: -(prevOffsetTop + 2)
+        };
+    };
+
+    updateSearchPluginsTableContextMenuOffset = function() {
+        // only re-calculate if window has moved
+        if ((prevOffsetLeft !== document.getElementById("searchPlugins").getBoundingClientRect().left) || (prevOffsetTop !== document.getElementById("searchPlugins").getBoundingClientRect().top))
+            searchPluginsTableContextMenu.options.offsets = calculateContextMenuOffsets();
+    };
+
+    setupSearchPluginTableEvents = function(enable) {
+        if (enable)
+            $$(".searchPluginsTableRow").each(function(target) {
+                target.addEventListener('dblclick', enablePlugin, false);
+                target.addEventListener('contextmenu', updateSearchPluginsTableContextMenuOffset, true);
+            });
+        else
+            $$(".searchPluginsTableRow").each(function(target) {
+                target.removeEventListener('dblclick', enablePlugin, false);
+                target.removeEventListener('contextmenu', updateSearchPluginsTableContextMenuOffset, true);
+            });
+    };
+
+    updateSearchPluginsTable = function() {
+        // clear event listeners
+        setupSearchPluginTableEvents(false);
+
+        var oldPlugins = Object.keys(searchPluginsTable.rows);
+        // remove old rows from the table
+        for (var i = 0; i < oldPlugins.length; i++) {
```

Comment:
use prefix operators: `++i`, and below too.

Reply:
...and in other files

Reply:
Will update per coding guidelines. Some of this code is almost a year old.